### PR TITLE
Don't rely on LLVM's -debug flag in tests

### PR DESCRIPTION
**Pull request by dschuff**
It's compiled away in release/non-asserts mode, so the invocation fails
with an unknown argument.

### CODE REVIEW

**Comment by dschuff** [comment]
@sbc100 I it doesn't *look* like this test is actually depending on debug output, is it? (it's really testing no-check-features, right?)

File: tests/test_other.py [hunk 1]
```diff
@@ -9086,7 +9086,7 @@ def test_supported_linker_flags(self):
     self.assertContained('warning: ignoring unsupported linker flag: `-rpath-link`', out)
 
     out = self.run_process([EMCC, path_from_root('tests', 'hello_world.cpp'),
-                            '-Wl,--no-check-features,-mllvm,-debug'], stderr=PIPE).stderr
```
Comment (sbc100) [inline tests/test_other.py:9089]: Probably just want to check multiple valid flags, can you just change `-debug` to any other llvm flag?

**Review by sbc100 — APPROVED** [review]
lgtm % comment

**Comment by dschuff** [comment]
This has been failing `test_run_wasi_sdk_output (test_other.other)`, but doesn't seem related at all. Is that a known issue?

**Comment by sbc100** [comment]
None of those test failures are related.  The simd ones should be fixed by #13400.   The wasi sdk test should have been fixed in #13383

**Comment by dschuff** [comment]
thanks for confirming, I'm going to land this.

**Pull request merged**